bigquery_etl/query_scheduling/dag.py

```diff
@@ -24,19 +44,93 @@ class InvalidDag(Exception):
     pass
 
 
+@attr.s(auto_attribs=True)
+class DagDefaultArgs:
+    """Representation of Airflow DAG default_args."""
+
+    owner: str = attr.ib()
```

Comment:
I'm not familiar with this `attr.ib()` approach which looks like it allows some special validation syntax? Can you add some code comments or additional info in the docstring here to explain the approach and link out to docs?

Reply:
I added doc comments to the classes that use `attrs`. It's a pretty neat library that makes defining classes a bit easier and reduces boilerplate code. It also has some cool features like validating attributes which comes in quite handy here.